content/en/docs/components/pipelines/author-a-pipeline/components.md

```diff
@@ -0,0 +1,222 @@
+
+    ```shell
+    kfp component build src/ --push-image
+    ```
+
+    For detailed information about all arguments/flags, see [CLI reference documentation](https://kubeflow-pipelines.readthedocs.io/en/master/source/cli.html#kfp-component-build).
+
+**When to use?** Containerized Python components should be used any time your component is implemented as Python code, but cannot be written as a standalone Python function or you wish to organize source code outside of the component Python function definition.
+
+### 3. Custom container components
+<!-- @zichuan-scott-xu to provide this section -->
+
+## Special case: Importer components
+Unlike the previous three authoring approaches, an importer component not a general authoring style but a pre-baked component for a specific use case: loading a machine learning [artifact](/) from remote storage to machine learning metadata (MLMD).
+
+**Before you continue:** Understand how KFP [Artifacts](/) work.
+
+Often times, the output artifact from one task is an input artifact to another task. In this case, the artifact can be easily accessed from the upstream task using `my_task.outputs['artifact_name']`. Additionally, the artifact is already registered in MLMD.
+
+But what if you wish to use an existing artifact that is not generated by a task in the current pipeline? Or wish to use as an artifact an external resource that was not generated by a pipeline at all? In either case, you would use an importer component to load an artifact from its URI.
+
+You do not need to write an importer component; it can be imported from the `dsl` module and used directly:
+
+```python
+from kfp import dsl
+
+@dsl.pipeline():
+def my_pipeline():
+    importer_task = dsl.importer(
+        artifact_uri='gs://ml-pipeline-playground/shakespeare1.txt',
+        artifact_class=dsl.Dataset)
+    other_component(dataset=importer_task.output)
+```
+
+In addition to the `artifact_uri`, you must provide an `artifact_class`.
+
+The `importer` component also permits setting artifact metadata via the `metadata` argument. You may also specify a boolean `reimport` argument. If `reimport` is `False`, KFP will use an existing MLMD artifact if it already exists from an earlier importer execution. If `reimport` is `True`, KFP will reimport the artifact as a new artifact, irrespective of whether it was previously imported.
+
+## Compile (save) a component
+Once you've written a component, you may wish to write the component definition to YAML for future use or submission for execution. This can be done via the KFP SDK DSL compiler:
+
+```python
+from kfp import compiler
+
+compiler.Compiler().compile(pipeline_func=addition_component, package_path='addition_component.yaml')
+```
+
```

Comment:
The previous doc contains a section of submitting a pipeline run via a client. I think it may help to keep that section for the reader to try creating their components themselves. WDYT? 
e.g. `client.create_run_from_pipeline_func()`

Reply:
Following on this, both of the above sample `train_model` components in the previous section have two inputs and an output. Maybe we can showcase how to call these components when submitting a run, i.e. only passing the inputs to the components but not the annotated outputs. This may be confusing to the beginners, but it could also be placed in the documentation about data passing.

Reply:
> The previous doc contains a section of submitting a pipeline run via a client.
I think it may make sense to include this in the "Submit a Pipeline Run" authoring section. It is also covered in the "Quickstart" section (PR #3317). WDYT?

> Maybe we can showcase how to call these components when submitting a run, i.e. only passing the inputs to the components but not the annotated outputs.

Good point on this and we definitely want to document this. I was planning on covering this in an authoring section called "Tasks" which describes how to instantiate and use tasks (PR #3322). WDYT?

Reply:
Thanks for clarifying! I agree that we should separate content like this and create a more linear reading experiences.